xls/contrib/integrator/ir_integrator.cc

```diff
@@ -166,6 +193,40 @@ IntegrationFunction::GetSourceFunctionIndexesOfNodesMappedToNode(
   return source_indexes;
 }
 
+absl::StatusOr<bool> IntegrationFunction::NodeSourceFunctionsCollide(
+    const Node* node_a, const Node* node_b) const {
+  // Get the index of the node's function if not owned by the integration
+  // function, otherwise get the source function indexes of the nodes mapped to
+  // the node.
+  auto get_node_source_indexes =
+      [this](const Node* node) -> absl::StatusOr<std::set<int64>> {
+    std::set<int64> sources;
+    if (IntegrationFunctionOwnsNode(node)) {
+      XLS_ASSIGN_OR_RETURN(sources,
```

Comment:
nit:I think  you can just return the result of GetSourceFunctionIndexesOfNodesMappedtoNode.

Reply:
Done.